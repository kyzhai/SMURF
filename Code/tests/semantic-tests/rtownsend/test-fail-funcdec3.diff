1,12c1
< Program: 
< 	SFuncdec: 
< 		f x :: Unknown = 3
< 	New Scope: 
< 		ID: x :: Unknown
< 
< 	
< Global Scope: 
< 	ID: f :: Int -> Bool -> Int
< 	ID: print :: Unknown
< 	ID: random :: Unknown
< 
---
> Fatal error: exception Sast.Pattern_num_mismatch(1, 2)
