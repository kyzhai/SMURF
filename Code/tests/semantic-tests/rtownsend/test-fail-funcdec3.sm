/* Number of patterns must match number of types in typesig minus one */

f :: Int -> Bool -> Int

f x = 3

main = f 0
