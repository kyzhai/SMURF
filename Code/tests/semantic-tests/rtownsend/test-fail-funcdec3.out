Program: 
	SFuncdec: 
		f x :: Unknown = 3
	New Scope: 
		ID: x :: Unknown

	
Global Scope: 
	ID: f :: Int -> Bool -> Int
	ID: print :: Unknown
	ID: random :: Unknown

